esrally/reporter.py

```diff
@@ -648,22 +648,23 @@ def _report_ingest_pipeline_failed(self, baseline_stats, contender_stats):
 
     def _report_disk_usage_stats_per_field(self, baseline_stats, contender_stats):
         best = {}
-        for index, total, field in total_disk_usage_per_field(baseline_stats):
-            best.setdefault(index, {})[field] = total
-        for index, total, field in total_disk_usage_per_field(contender_stats):
-            for_idx = best.setdefault(index, {})
-            prev = for_idx.get(field, 0)
-            if prev < total:
-                for_idx[field] = total
+        if baseline_stats.disk_usage_total:
```

Comment:
should we only perform this work at all if both have `disk_usage_total`?  so instead of these changes do (in line 415)
```python
if baseline_stats.disk_usage_total and contender_stats.disk_usage_total:
    metrics_table.extend(self._report_disk_usage_stats_per_field(baseline_stats, contender_stats))

```

Reply:
That makes sense. Can we issue a warning about it? Something on the console?

Reply:
> That makes sense. Can we issue a warning about it? Something on the console?

I was ready to add a warning until I thought it would add noise when running comparisons against metrics generated with older Rally versions and `disk_usage_total=[]` for later versions. I'm interested in your thoughts.

What do you think instead about setting `<x>.disk_usage_total=[]` when `<x>.disk_usage_total=None`? This way we'll show disk usage metrics if they exist for one of the subjects, and not leave anything out.